### PR TITLE
added rpc for get mining status

### PR DESCRIPTION
**Pull request by antonlerner**
Reponses are : post init status, coinbase reward account and post data dir path

### CODE REVIEW

File: activation/activation.go [hunk 1]
```diff
@@ -246,6 +247,7 @@ func (b *Builder) StartPost(rewardAddress address.Address, dataDir string, space
 	b.SetCoinbaseAccount(rewardAddress)
 	b.postInitLock.Lock()
 	if b.initStatus == Done {
+		b.postInitLock.Unlock()
```
Comment (y0sher) [inline activation/activation.go:250]: @moshababo did some changes here so please make sure it doesn't collapse

File: nipst/nipst.go [hunk 2]
```diff
@@ -228,12 +228,22 @@ func (nb *NIPSTBuilder) IsPostInitialized() bool {
 	return nb.postProver.Initialized()
 }
 
+func (nb *NIPSTBuilder) GetDataDirPath() string {
+	var d string
+	nb.cfgM.RLock()
```
Comment (zalmen) [inline nipst/nipst.go:233]: why do you need this lock? Where is it modified (after first initialization)?

Comment (antonlerner) [reply nipst/nipst.go:233]: at initialization, this does not happen in a certain order and potentially can happen in the same time, hence the lock

File: api/api_test.go [hunk 3]
```diff
@@ -405,8 +409,8 @@ func TestJsonWalletApi(t *testing.T) {
 	got, want = resp.StatusCode, http.StatusOK
 	assert.Equal(t, want, got)
 
-	// test get eligible layers / miner rewards
-	url = fmt.Sprintf("http://127.0.0.1:%d/v1/startmining", config.ConfigValues.JSONServerPort)
+	// test get statistics about init progress
+	url = fmt.Sprintf("http://127.0.0.1:%d/v1/stats", config.ConfigValues.JSONServerPort)
```
Comment (y0sher) [inline api/api_test.go:413]: we don't want to test startmining anymore ?

File: api/pb/api.proto [hunk 4]
```diff
@@ -97,39 +103,15 @@ service SpacemeshService {
...
-        option (google.api.http) = {
-          post: "/v1/gettotalawards"
-          body: "*"
-        };
-    }
-    rpc GetUpcomingAwards(google.protobuf.Empty) returns (EligibleLayers) {
+    rpc GetMiningStats(google.protobuf.Empty) returns (MiningStats) {
         option (google.api.http) = {
-          post: "/v1/getupcomingawards"
+          post: "/v1/stats"
```
Comment (y0sher) [inline api/pb/api.proto:114]: mining-stats ?

**Review by y0sher — APPROVED** [review]

File: api/api_test.go (hunk 3 above)
Comment (antonlerner) [reply api/api_test.go:413]: I have a test for that before this one, that was a duplicate

File: api/pb/api.proto (hunk 4 above)
Comment (antonlerner) [reply api/pb/api.proto:114]: let leave it as is for now, there are no other stats or other stats can be added to this rpc later on

**Pull request merged**